toolchain/lower/handle_aggregates.cpp

```diff
@@ -141,9 +142,12 @@ static auto EmitAggregateInitializer(FunctionContext& context,
       // non-constant initialization.
       for (auto [i, ref_id] : llvm::enumerate(refs)) {
         if (context.sem_ir().constant_values().Get(ref_id).is_constant()) {
-          auto init =
-              context.sem_ir().insts().GetAs<SemIR::InitializeFrom>(ref_id);
-          HandleInst(context, ref_id, init);
+          auto dest_id =
+              SemIR::FindReturnSlotArgForInitializer(context.sem_ir(), ref_id);
```

Comment:
This is a question for @zygoloid but I feel like we generally want the SemIR to express stuff and lower to be less clever. Should Check be ensuring that we end up with an InitializeFrom instruction as the output of Convert (or something above Convert) here?